dev-tools/packaging/templates/darwin/elastic-agent.tmpl

```diff
@@ -6,6 +6,9 @@ set -e
 symlink="/Library/Elastic/Agent/elastic-agent"
 
 if test -L "$symlink"; then
-    ln -sfn "data/elastic-agent-{{ commit_short }}/elastic-agent.app/Contents/MacOS/elastic-agent" "$symlink"
+    symlinkTarget="data/elastic-agent-{{ commit_short }}/elastic-agent.app/Contents/MacOS/elastic-agent"
+    if test -f "data/elastic-agent-{{ agent_package_version }}{{ snapshot_suffix }}-{{ commit_short }}/elastic-agent.app/Contents/MacOS/elastic-agent"; then
```

Comment:
Out of curiosity why is this necessary? I though the package structure was static at the point of packaging?

Reply:
This is the script that fixes the links to the elastic-agent executable within the `elastic.app` directory. Now the path has potentially changed depending on how the package has been installed so the symlink target needs to be tested and set to the existing path

Reply:
Ah I missed that this script was executed at runtime, thanks. Thought it was part of the packaging process itself.